app/services/delete_reference.rb

```diff
@@ -0,0 +1,7 @@
+class DeleteReference
+  def call(reference:)
+    raise 'Application has been sent to providers' unless reference.application_form.application_choices.all?(&:unsubmitted?)
```

Comment:
Can use `application_form.submitted?` here. It's a different check (looks for submitted_at timestamp), but sufficient I think.